src/GrpcProtos/capture.proto

```diff
@@ -97,7 +97,7 @@ message InternedCallstack {
 }
 
 message CallstackSample {
-  int32 pid = 1;
+  int32 pid = 1;  // Note that pid is currently not set as the client doesn't need it.
```

Comment:
maybe we should fix this...

Reply:
or if it is really by design, remove this field?

Reply:
if it is easy to set we should probably set it and just have it here ready for use if needed.

Reply:
It then just gets dropped by CaptureEventProcessor as `orbit_client_protos::CallstackEvent` doesn't have a process id. Do we want to do it anyway? Maybe. Do we want to add a process id to `CallstackEvent`? Maybe. In the context of this change I don't know and it's out of scope.

Reply:
you have discovered this while writing the test, so I would say it is in the scope. I would say we should set it here. If then client ignores it it is fine. This proto will become persistent soon which means that it could become used in the latter version of orbit.

Reply:
> you have discovered this while writing the test, so I would say it is in the scope.

Why? It was intended behavior.

> This proto will become persistent soon which means that it could become used in the latter version of orbit.

That's why this is out of scope. We should decide what fields we want set and what we don't separately to prepare for a stable version. And that includes the other protos that have the same property (`ThreadName`, `ThreadStateSlice`). I'm not saying we shouldn't set process ids, I'm saying it's not a decision for this change.